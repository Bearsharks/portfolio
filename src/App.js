import './App.scss';
import { BrowserRouter as Router, Route, NavLink, Switch, Redirect } from "react-router-dom";
import Home from "./Home"
//다음목표 about 페이지 만들기
//about을 채워보기
//인터액션 
//클립파일 스크롤하면 다음페이지로 넘어감 클릭해서 잡아끌면 다음페이지로 넘어감 
//스크롤할때 섹션위치에 따라 네비게이션이 바뀐다.
//네비게이션을 클릭하면 해당 위치로 이동한다.

function App() {
	return (
		<div className="App">
			<header className="App-header">
			</header>

			<Router>
				<NavLink to={`/portfolio/Home`} activeClassName={'linkActive'}>Home</NavLink>
				<NavLink to={`/portfolio/work`} activeClassName={'linkActive'}>work</NavLink>
				<NavLink to={`/portfolio/contact`} activeClassName={'linkActive'}>contact</NavLink>
				<NavLink to={`/portfolio/youtube`} activeClassName={'linkActive'}>youtube</NavLink>
				<Switch>
					<Route exact path="/portfolio">
						<Redirect to={`/portfolio/Home`} />
					</Route>
					<Route path={`/portfolio/Home`}>
						<Home></Home>
					</Route>
					<Route path={`/portfolio/work`}>
						<div>2</div>
					</Route>
					<Route path={`/portfolio/contact`}>
						<div>3</div>
					</Route>
					<Route path={`/portfolio/youtube`}>
						<div>4</div>
					</Route>
				</Switch>
			</Router>
			<footer>

			</footer>
		</div >
	);
}

export default App;
