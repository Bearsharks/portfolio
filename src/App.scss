
.App{
    margin: auto;
    overflow:hidden;
    height: 100vh;
}

header{
    height: 50px;
}

.header-wrapper {
    display: flex;
    float:right;
    align-items: center;
    justify-content: space-between;
    height: 50px;
    width: 100%;
}
.home-btn{
    z-index: 100;
}
main {
    height: calc(100% - 50px);
}

.grecaptcha-badge{
    visibility: hidden;
}