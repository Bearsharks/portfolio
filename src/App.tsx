import Intro from 'components/Intro';
import MyFooter from 'components/MyFooter';
import Projects from 'MainContents';
import Modal from 'popups/Modal';
import './App.scss';

function App() {
  return (
    <div className="App">
      <main>
        <header>
          <h1>
            Portfoilo
          </h1>
        </header>
        <Intro></Intro>
        <Projects></Projects>
        <footer>
          <MyFooter />
        </footer>
      </main>
      <Modal></Modal>
    </div>
  );
}

export default App;
