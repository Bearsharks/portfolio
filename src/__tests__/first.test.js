// import dependencies
import React from 'react'

// import API mocking utilities from Mock Service Worker
import { rest } from 'msw'
import { setupServer } from 'msw/node'

// import react-testing methods
import { render, fireEvent, waitFor, screen } from '@testing-library/react'

// add custom jest matchers from jest-dom
import '@testing-library/jest-dom/extend-expect'

import App from "../App"
describe('app test', () => {
    it('renders component correctly', () => {
        const { container } = render(<App />);
        const el = screen.getByText('home버튼');
        expect(el).toBeInTheDocument();
    });
    it('hbg work', () => {
        const { container } = render(<App />);
        const navi = container.querySelector('.navigation');
        expect(navi).toHaveClass("navigation--inactive");
        let hbg = screen.getByRole('button');
        fireEvent.click(hbg);
        expect(navi).toHaveClass("navigation--active");
        fireEvent.click(hbg);
        expect(navi).toHaveClass("navigation--inactive");

    })
});