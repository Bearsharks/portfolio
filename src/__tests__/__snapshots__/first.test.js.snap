// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`처음 테스트 renders component correctly 1`] = `
<div>
  <div
    class="App"
  >
    <header>
      <div
        class="header-wrapper"
      >
        <div
          class="home-btn"
        >
          home버튼
        </div>
        <div
          class="hamburger-btn"
        >
          <div
            class="hamburger"
          />
        </div>
      </div>
    </header>
    <main>
      <div
        class="navigation navigation--inactive"
      >
        <a
          href="/portfolio/home"
        >
          Home
        </a>
        <a
          href="/portfolio/about"
        >
          About
        </a>
        <a
          href="/portfolio/works"
        >
          Works
        </a>
        <a
          href="/portfolio/contact"
        >
          Contact
        </a>
      </div>
    </main>
  </div>
</div>
`;
